Update contracts.rst

mocamircea opened the pull request:
Extended example according to the discussed stuff, clarification & typos

mocamircea committed:
Update contracts.rst

Extended example according to the discussed stuff, clarification & typos

chriseth commented on docs/contracts.rst:
```diff
@@ -194,9 +194,16 @@ return parameter list for functions.
+    contract Caller {
+        function readData(){
+            C c = new C();
+            uint local = c.data(); // call accessor
```
Accessors have not been introduced at this point.

mocamircea replied:
yes but lines 198-200 rely on c.data() call which is related to accessor

mocamircea replied:
Solution: a link to the Accessor Functions section below OR completely remove the comment?

chriseth replied:
It is probably better to remove any mention of accessor functions here (instead use regular functions) and then explain the modifiers that are relevant for accessor functions in the section about accessor functions.

mocamircea committed:
Update contracts.rst

I think it's more clear now and without the risk of mentioning the yet discussed accessor concept

chriseth commented on docs/contracts.rst:
```diff
@@ -194,9 +194,16 @@ return parameter list for functions.
         function setData(uint a) internal { data = a; }
         uint public data;
```
As discussed, please remove the `public` from the variable here and change to:

```
   function setData() internal { _data = a; }
   function getData() { return _data; }
   uint _data;
```

mocamircea committed:
Update contracts.rst

I left the example with contract C only, showing the access specifiers discussed at 186-188. The specifier of data is left public since nothing is related to accessor functions now.

Added a separate example with contracts C, D and E to show specifier's effect. 

Added a separate example to prove the synthesizing of accessor functions.

Added a new example to show the internal/external call of the accessor, according to existent discussion.
